packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -212,12 +219,28 @@ const serviceMapFactory = (options: ServiceMapFactoryOptions) => {
           cardanoNode,
           dbPools,
           logger,
-          ntfMetadataService,
+          ntfMetadataService: dbSyncNftMetadataService,
           tokenMetadataService
         }
       );
 
-      return new AssetHttpService({ assetProvider, logger });
+      const typeormNftDependencies = {
+        connectionConfig$: connectionConfigs.Handle!,
+        entities: getEntities(['asset']),
+        logger
+      };
+
+      const typeormNftMetadataService = new TypeOrmNftMetadataService(typeormNftDependencies);
+
+      const typeormAssetProvider = new TypeormAssetProvider({
+        ...typeormNftDependencies,
+        nftMetadataService: typeormNftMetadataService
+      });
+
+      return new AssetHttpService({
+        assetProvider: args.useTypeormNftMetadata ? typeormAssetProvider : dbSyncAssetProvider,
```

Comment:
The problem here is that we are going to create both the `HttpService`s but we are going to use only one pf them.
Each `HttpService` requires its specific configuration variables (about DB connection), so this way we are going to require the cfg for both the `HttpService`s regardless from which one we want to use.

It's better to put under two distinct `if` branches also the creation on the `HttpService`s.

Reply:
I think you can just follow the pattern of how it works with `useTypeormStakePoolProvider`

Reply:
Fixed on 2e7a67a